packages/agoric-cli/test/test-inter-cli.js

```diff
@@ -82,7 +82,7 @@ const makeNet = published => {
   const ctx = makeFromBoard();
   const m = boardSlottingMarshaller(ctx.convertSlotToVal);
   const fmt = obj => {
-    const capData = m.serialize(obj);
+    const capData = m.serialize(harden(obj));
```

Comment:
Based on previous answer, resolving.

Reply:
Do we have a convention about _where_ to `harden()`? In this case, the source of the mutable objects are calls like `makeNet({ ...publishedNames, wallet: govWallets })`, which returns a `fetch` mock that itself returns an object with a `json` method returning the result of calling `fmt` on a value reachable by descending that initial structure—and it is only here where that [deep] value is hardened, right before asking a marshaller to encode it. It seems like the call would be more appropriate before invoking `makeNet`, or else right at the top of `makeNet` (e.g., to avoid prettier-induced trauma).

Reply:
Outside of test code, the convention is, as early as possible, which almost always means at the point where the object is first created. Which often means, the object/array/regexp/function literal expression that evaluates to the object.

Inside test code, we're more relaxed, but still try to follow this when not inconvenient.

What's trickier is conventions about when not to use harden. We have too many hardens in places they shouldn't be, that mask bugs of the absence of earlier hardens. 

But over time we've gotten more relaxed about those, which I think is ok for now. Truly, we're never gonna get this right until we develop and use Jessie static checking rules with soundness properties. Until we have static checking of these, we will always get them wrong, and we shouldn't put ***too much*** effort into stopgaps short of developing those static checks. But we should still flag obvious violations during code review. We should spend more effort flagging missing hardens that should be added than masking hardens that should be removed. Attn @michaelfig

Reply:
> ... Truly, we're never gonna get this right until we develop and use Jessie static checking rules with soundness properties.

For reference: I tend to keep my research notes on that in https://github.com/agoric-labs/jessica/issues/35

Speaking of which... @jake-gillberg, you recently asked how to dig in to the Agoric platform. This would be a _fantastic_ contribution. _Let's follow up over there and/or in discord._

Reply:
> This would be a fantastic contribution.

In would indeed!